src/cats/BreedSelector.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useMemo } from "react";
+import clsx from "clsx";
+import { Status } from "./useAsync";
+import styles from "./BreedSelector.module.css";
+
+type BreedType = {
+  id: string;
+  name: string;
+};
+
+type BreedSelectorPropsType = {
+  onSelectBreedID: (id: string) => void;
+  breeds: Array<BreedType> | null;
+  selectedBreedID: string;
+  status: Status;
+};
+
+export default function BreedSelector({
+  onSelectBreedID,
+  breeds,
+  selectedBreedID,
+  status,
+}: BreedSelectorPropsType) {
+  const isError = status === "error";
+  const onSelectChange = useCallback(
+    (e: React.ChangeEvent<HTMLSelectElement>) => {
+      onSelectBreedID(e.target.value);
+    },
+    [onSelectBreedID]
+  );
+
+  const selectOptions = useMemo(() => {
```

Comment:
just like with memoizing callbacks, you don't need to memoize the children passed to a primitive like `select`

Reply:
Thanks, this is good to know.

Reply:
so this useMemo is trying to avoid the work to recreate the options array on every render, not for stability.